docs/pages/image-extractor.rst

```diff
@@ -98,77 +115,478 @@ below is not represented.
+            plt.figure(figsize=(12, 8))
+            for i, img in enumerate(imgs):
+                ax = plt.subplot(1, batch_size, i + 1)
+                ax.axis("off")
+                if img_type == 'rgb':
+                    plt.imshow(img.numpy().transpose(1, 2, 0))
+                elif img_type in ['truth', 'prediction']:
+                    plt.imshow(img.numpy())
+
+            plt.show()
+
+        batch_size = len(sample_batch['rgb'])
+        for k in sample_batch.keys():
+            show_row(sample_batch[k], batch_size, k)
+
+
+    with torch.no_grad():
+        model.to('cpu')
+        model.eval()
+        _, batch = next(enumerate(dataloader))
+        mask_pred = model(batch['rgb'])
+        mask_pred = F.softmax(mask_pred, dim=1)
+        mask_pred = torch.argmax(mask_pred, dim=1)
+
+        batch['prediction'] = mask_pred
+
+        show_batch(batch)
+
+
+.. image:: ../images/semantic-segmentation-results.png
+
+
+On the top row we see the input to the model which is the batch of RGB images. On the middle row is the grouth truth masks. On the bottom row are the masks that the model predicted.
+
+
+
+.. _Replica dataset: https://github.com/facebookresearch/Replica-Dataset
+.. _Pytorch Dataset and Dataloader: https://pytorch.org/tutorials/beginner/data_loading_tutorial.html
+.. _installation instructions: https://github.com/facebookresearch/habitat-sim
+.. _transform: https://pytorch.org/docs/stable/torchvision/transforms.html
+.. _UNET: https://arxiv.org/abs/1505.04597
+.. _this github repo: https://github.com/milesial/Pytorch-UNet
+.. _RMSprop: http://www.cs.toronto.edu/~tijmen/csc321/slides/lecture_slides_lec6.pdf
+
+
+
+`Writing Custom Pose Extractors`_
+=================================
+
+Each instance of an ImageExtractor has a pose extractor (an instance of PoseExtractor). The pose extractor defines how camera poses are programmatically determined so that the image
+extractor knows how to manipulate the camera position and angle to extract an image from habitat
+
+Users can write their over subclass of PoseExtractor to define custom ways of getting these camera poses. All custom pose extractors must inherit from the PoseExtractor abstract class and
+override the extract_poses method. Further, the user must register the pose extractor using
+habitat_sim.registry (i.e. adding the @registry.register_pose_extractor(name) decorator). This allows you to pass the name of your custom pose extractor to the ImageExtractor constructor. For more detailed examples of using the Habitat registry, see `this code`_.
+
+`Default Behavior`_
+-------------------
+
+The default behavior is reliant on something called the topdown view of a scene, which is just a two-dimensional birds-eye representation of the scene. The topdown view is a two-dimensional array of 1s and 0s where 1 means that pixel is "navigable" in the scene (i.e. an agent can walk on top of that point) and 0 means that pixel is "unnavigable".
```

Comment:
This is good. Put in links to the ECCV20 tutorial Sim-Basics Colab where users can find out more about the NavMesh and how to compute top-down maps from Hab-Sim.